contrib/babelfishpg_tsql/src/tsqlIface.cpp

```diff
@@ -4060,6 +4072,10 @@ makeSetStatement(TSqlParser::Set_statementContext *ctx, tsqlBuilder &builder)
 			auto option = set_special_ctx->set_on_off_option().front();
 			if (option->BABELFISH_SHOWPLAN_ALL())
 				return makeSetExplainModeStatement(ctx, true);
+            // PARSEONLY needs to be able to be turned off once it is on, so it is handled at parse time.
```

Comment:
According to the comment, we should check first whether PARSEONLY is already on.

Reply:
I think it makes no difference, right? If it is off, then flipping the bool shouldn't have any effect